### PR TITLE
Match: support for supplying multiple app identifiers at once

### PR DESCRIPTION
**Pull request by jverdi**
🔑 Allows you to supply multiple app identifiers to match at once and it will only perform (1) clone of your repo, and (1) check for your associated certificate.

As an action:

```
match(type: "appstore", 
      app_identifier: ["tools.fastlane.app", "tools.fastlane.app2", "tools.fastlane.app3"])
```

I made sure to also continue supporting the old syntax, so this doesn't break existing configurations:

```
match(type: "appstore", app_identifier: "tools.fastlane.app")
```

Multiple apps via command-line:

```
match appstore -a tools.fastlane.app,tools.fastlane.app2,tools.fastlane.app3
```

This addresses performance issues discussed in #4100

I don't see the need for a full matrix-like solution. Generally speaking, CI builds will be building for one environment at a time. The real pain point has been the proliferation of app ids for extensions. A matrix also doesn't translate well for command-line usage.

### CODE REVIEW

File: match/lib/match/runner.rb [hunk 1]
```diff
@@ -10,25 +10,47 @@ def run(params)
...
       if self.changes_to_commit and !params[:readonly]
         message = GitHelper.generate_commit_message(params)
         GitHelper.commit_changes(params[:workspace], message, params[:git_url], params[:git_branch])
       end
 
-      TablePrinter.print_summary(params)
+      # Print a summary table for each app_identifier
+      app_identifiers.each do |app_identifier|
+        params[:app_identifier] = app_identifier
+        uuid = uuids[app_identifier]
```
Comment (jverdi) [inline match/lib/match/runner.rb]: fixed

**Comment by mathiasAichinger** [comment]
Looking forward to this! 👍

**Comment by hjanuschka** [comment]
ohhh my gosh! i so much need that!  - let me know if you need some help anywhere.

**Comment by KrauseFx** [comment]
Hey @jverdi, thanks for the pull request, sorry it took so long starting to review this, will do that now 👍

**Comment by KrauseFx** [comment]
Awesome, thanks @jverdi, I resolved the merge conflicts and moved it over to https://github.com/fastlane/fastlane/pull/6635 👍

**Pull request closed without merging**